torch/testing/_internal/common_methods_invocations.py

```diff
@@ -145,6 +145,104 @@ def _getattr_qual(obj, name, default=_NOTHING):
         else:
             raise
 
+# NOTE: OpInfos and the OpInfo testing pattern
+# See also: https://github.com/pytorch/pytorch/wiki/Writing-tests-in-PyTorch-1.8
+# See also: https://github.com/pytorch/pytorch/issues/54261, which tracks
+#   which OpInfos we still want to create
+#
+# An "OpInfo", described by the class and subclasses below, is a collection
+#   of metadata related to a PyTorch operator, like torch.add. This metadata
+#   can be separated into three categories:
+#
+#     1) Metadata describing the operator, like "dtypesIfCPU", which describes
+#          the dtypes the operator supports on the CPU. This metadata
+#          is used by generated tests to understand what they should test
+#          and what the expected results of the test should be.
+#     2) Test directives, like "skips", which indicates that some generated
+#          tests should be skipped for this operator. Test directives
+#          directly control the behavior of generated tests.
+#     3) A "sample_inputs_func", a function that can generate valid inputs to
```

Comment:
sample_inputs_func deserves more comments - does it provide several inputs cases? How one should go about deciding how many input variants to add? How are kwargs handled? How can inputs for inplace operation variants be filtered?